pkg/controller/serviceaccount/tokens_controller.go

```diff
@@ -520,3 +656,50 @@ func getSecretReferences(serviceAccount *api.ServiceAccount) sets.String {
 	}
 	return references
 }
+
+func makeServiceAccountKey(sa *api.ServiceAccount) string {
+	return fmt.Sprintf("%s/%s/%s", sa.Namespace, sa.Name, sa.UID)
+}
+
+func parseServiceAccountKey(key interface{}) ( /*ns*/ string /*name*/, string /*uid*/, types.UID, error) {
```

Comment:
That would be much better, of course.

On Tue, Apr 5, 2016 at 11:55 AM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/controller/serviceaccount/tokens_controller.go
> <https://github.com/kubernetes/kubernetes/pull/23858#discussion_r58595522>
> :
>
> > @@ -520,3 +656,50 @@ func getSecretReferences(serviceAccount *api.ServiceAccount) sets.String {
> >  	}
> >  	return references
> >  }
> > +
> > +func makeServiceAccountKey(sa *api.ServiceAccount) string {
> > +	return fmt.Sprintf("%s/%s/%s", sa.Namespace, sa.Name, sa.UID)
> > +}
> > +
> > +func parseServiceAccountKey(key interface{}) ( /*ns*/ string /*name*/, string /*uid*/, types.UID, error) {
>
> I'm actually leaning toward making a struct as my key instead of munging
> everything into a string and then parsing it back out
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23858/files/b8f98d9a14e121eb243af96611a9ecb304179bb9#r58595522>
>